core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -384,6 +384,14 @@ public void testHavingReferencesOutputAlias()
                 .hasMessage("line 1:32: Column 'x' cannot be resolved");
     }
 
+    @Test
+    public void testRowReferencesUnknownField()
+    {
+        assertFails("SELECT row('a', 'b', 'c').field")
+                .hasErrorCode(INVALID_COLUMN_REFERENCE)
```

Comment:
INVALID_COLUMN_REFERENCE seems wrong. Do we need new error code.
cc: @martint

Reply:
I've added a new error type. @martint ptal

Reply:
The INVALID_COLUMN_REFERENCE error code is fine for this purpose. Fields and columns are two ways to look at the same thing. Precisely, a table is a `multiset` of values of type `row`, so top level columns *are* fields.

Reply:
Will revert then